packages/carbon-graphs/src/js/helpers/axis.js

```diff
@@ -893,7 +893,25 @@ const createAxes = (axis, scale, config, canvasSVG) => {
       )})`,
     )
     .call(axis.x)
-    .call(resetD3FontSize);
+    .call(resetD3FontSize)
+    .attr('id', 'x')
+    .selectAll('text')
+    .style('text-anchor', () => {
+      if (!utils.isDefined(config.axis.x.ticks.tickLabelsRotation)) {
+        config.axis.x.ticks.tickLabelsRotation = 0;
+        return 'middle';
+      }
+      const rotation = config.axis.x.ticks.tickLabelsRotation;
+      if (rotation === 0) {
+        return 'middle';
+      } if (rotation === -45) {
```

Comment:
Hardcoding -45 defeats the purpose of utilizing the set array. We should minimize refactoring done in the future if we need to add more angles. The conditions should be:

tickLabelsRotation = 0 -> return middle
else return end

tickLabelsRotation is undefined and tickLabelsRotation is invalid value -> tickLabelsRotation should be covered by the logic added in Graph.js.